### PR TITLE
improving analysis

### PR DESCRIPTION
**Pull request by oddcoder**
improving for arm
removing argument detection extension used in af now args are seperately detected by afCa
aa = af + afCa
using aliases as well as real registers
rewritten varsub
Fixing some of Arm esil and optimizing them for arg/var detection

### CODE REVIEW

File: libr/anal/fcn.c [hunk 1]
```diff
@@ -532,26 +536,11 @@ static int fcn_recurse(RAnal *anal, RAnalFunction *fcn, ut64 addr, ut8 *buf, ut6
 			break;
 		// TODO: use fcn->stack to know our stackframe
 		case R_ANAL_STACK_SET:
-			if ((int)op.ptr > 0) {
```
Comment (radare) [inline libr/anal/fcn.c]: uhm removing this is gonna break a lot of things

Comment (radare) [inline libr/anal/fcn.c]: maybe make it optional?

Comment (radare) [inline libr/anal/fcn.c]: or disabled depending on the anal plugin? (adding a new field in the RAnalPlugin struct)

Comment (oddcoder) [inline libr/anal/fcn.c]: or better fix what is broken in fact If the broken code is more than what I can handle I can just do the merge of af and afCa but still this code it already breaks too much to stay

**Comment by oddcoder** [comment]
WIP

**Comment by radare** [comment]
Can you rebase to master? I have updated anal.arm64

**Comment by radare** [comment]
rebase again plz O:)

maybe we should ask github to add a "rebase" button cc @Maijin @XVilka

**Comment by radare** [comment]
arm regression fixed and tests added, waiting for travis

**Comment by oddcoder** [comment]
I am also done here

**Comment by Maijin** [comment]
Not yet:

```

[35mRegression: [0mrun_tests.sh
[35mCommand: [0m /home/travis/build/radare/radare2/install/bin/radare2 -e scr.color=0 -N -q -i /tmp/r2-regressions//vars-Rl2WWS/rad   ../../bins/elf/armcall > /tmp/r2-regressions//vars-Rl2WWS/out 2> /tmp/r2-regressions//vars-Rl2WWS/err
[35mFile: [0m../../bins/elf/armcall
[35mScript: [0m
afr@main
pdf@sym.call

[35mDiff: [0m--- /tmp/r2-regressions//vars-Rl2WWS/exp	2016-06-17 18:16:05.160012946 +0000
+++ /tmp/r2-regressions//vars-Rl2WWS/out	2016-06-17 18:16:05.172013385 +0000
@@ -5,9 +5,9 @@
 |           0x00010420      00482de9       push {fp, lr}
 |           0x00010424      04b08de2       add fp, sp, 4
 |           0x00010428      08d04de2       sub sp, sp, 8
-|           0x0001042c      08000be5       str r0, [fp - arg_8h]
+|           0x0001042c      08000be5       str r0, [fp, -8]
 |           0x00010430      14009fe5       ldr r0, [pc, 0x14]          ; [0x1044c:4]=0x104f0 loc._d_14 LEA loc._d_15 ; loc._d_15
-|           0x00010434      08101be5       ldr r1, [fp - arg_8h]
+|           0x00010434      08101be5       ldr r1, [fp, -8]
 |           0x00010438      a2ffffeb       bl sym.imp.printf
 |           0x0001043c      0030a0e3       mov r3, 0
 |           0x00010440      0300a0e1       mov r0, r3
```

**Comment by oddcoder** [comment]
I have them on my machine fixed 
but cant merge fixed tests till branch is merged

**Comment by oddcoder** [comment]
https://github.com/radare/radare2-regressions/pull/463

**Comment by oddcoder** [comment]
fixing memory leaks in r_anal_var_get_byname 
related to https://github.com/radare/radare2/issues/5165

**Pull request merged**